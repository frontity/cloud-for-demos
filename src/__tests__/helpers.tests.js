const { getIdFromClass, getSlugFromSrc } = require('../helpers');

const imagesWithClass = require('./data/imagesWithClass.json');
const imagesWithClassButNoId = require('./data/imagesWithClassButNoId.json');
const imagesWithDataSrc = require('./data/imagesWithDataSrc.json');
const imagesWithSrc = require('./data/imagesWithSrc.json');
const imagesWithDataOriginal = require('./data/imagesWithDataOriginal.json');
const imagesWithSrcSizes = require('./data/imagesWithSrcSizes.json');

describe('Helper getIdFromClass', () => {
  test('Should return `null` if no id are extracted from class', () => {
    imagesWithDataSrc.forEach(image => {
      const id = getIdFromClass(image);
      expect(id).toBe(null);
    });

    imagesWithClassButNoId.forEach(image => {
      const id = getIdFromClass(image);
      expect(id).toBe(null);
    });
  });
  test('Should return an id', () => {
    imagesWithClass.forEach(image => {
      const id = getIdFromClass(image);
      expect(typeof id).toBe('number');
    });
  });
});

describe('Helper getSlugFromSrc', () => {
  test('Should return `null` if no slug are extracted', () => {
    imagesWithClass.forEach(image => {
      const slug = getSlugFromSrc(image);
      expect(slug).toBe(null);
    });
  });
  test('Should return a slug extracted from data-src', () => {
    imagesWithDataSrc.forEach(image => {
      const slug = getSlugFromSrc(image);
      expect(typeof slug).toBe('string');
    });
  });
  test('Should return a slug extracted from src', () => {
    imagesWithSrc.forEach(image => {
      const slug = getSlugFromSrc(image);
      expect(typeof slug).toBe('string');
    });
  });
  test('Should return a slug extracted from data-original', () => {
    imagesWithDataOriginal.forEach(image => {
      const slug = getSlugFromSrc(image);
      expect(typeof slug).toBe('string');
    });
  });
  test('Should return the right slug if the filename has a size', () => {
    const slugs = [
      'image_11857',
      'image_11858x',
      'image_11861-',
      'image_11859_',
      'image_11860---',
    ];

    imagesWithSrcSizes.forEach((image, index) => {
      const slug = getSlugFromSrc(image);
      expect(typeof slug).toBe('string');
      expect(slug).toBe(slugs[index]);
    });
  });
});
