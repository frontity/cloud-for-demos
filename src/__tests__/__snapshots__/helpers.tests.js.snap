// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper getArrayOfImages Should return a flat array of images 1`] = `
Array [
  Object {
    "attributes": Array [
      Object {
        "key": "class",
        "value": "alignnone size-full wp-image-11857",
      },
      Object {
        "key": "src",
        "value": "https://frontity.com/wp-content/uploads/image_11857.jpg",
      },
      Object {
        "key": "srcset",
        "value": "https://frontity.com/wp-content/uploads/image_11857.jpg 600w, https://frontity.com/wp-content/uploads/image_11857-300x200.jpg 300w",
      },
    ],
    "children": Array [],
    "tagName": "img",
    "type": "element",
  },
  Object {
    "attributes": Array [
      Object {
        "key": "class",
        "value": "alignnone size-full wp-image-11858",
      },
      Object {
        "key": "src",
        "value": "https://frontity.com/wp-content/uploads/image_11858.jpg",
      },
      Object {
        "key": "srcset",
        "value": "https://frontity.com/wp-content/uploads/image_11858.jpg 600w, https://frontity.com/wp-content/uploads/image_11858-300x200.jpg 300w",
      },
    ],
    "children": Array [],
    "tagName": "img",
    "type": "element",
  },
  Object {
    "attributes": Array [
      Object {
        "key": "class",
        "value": "alignnone size-full wp-image-11861",
      },
      Object {
        "key": "src",
        "value": "https://frontity.com/wp-content/uploads/image_11861.jpg",
      },
    ],
    "children": Array [],
    "tagName": "img",
    "type": "element",
  },
  Object {
    "attributes": Array [
      Object {
        "key": "class",
        "value": "alignnone size-full wp-image-11859",
      },
      Object {
        "key": "src",
        "value": "https://frontity.com/wp-content/uploads/image_11859.jpg",
      },
    ],
    "children": Array [],
    "tagName": "img",
    "type": "element",
  },
  Object {
    "attributes": Array [
      Object {
        "key": "class",
        "value": "alignnone size-full wp-image-11860",
      },
      Object {
        "key": "src",
        "value": "https://frontity.com/wp-content/uploads/image_11860.jpg",
      },
    ],
    "children": Array [],
    "tagName": "img",
    "type": "element",
  },
]
`;

exports[`Helper getArrayOfImages Should return an empty array if no images are provided 1`] = `Array []`;

exports[`Helper getIdsFromClass Should return an array of ids 1`] = `
Array [
  11857,
  11858,
  11861,
  11859,
  11860,
]
`;

exports[`Helper getIdsFromClass Should return an empty array if no ids are extracted from class 1`] = `Array []`;

exports[`Helper getIdsFromClass Should return an empty array if no ids are extracted from class 2`] = `Array []`;

exports[`Helper getSlugsFromSrc Should return an array of slugs extracted from data-original 1`] = `
Array [
  "image_11857",
  "image_11858",
  "image_11861",
  "image_11859",
  "image_11860",
]
`;

exports[`Helper getSlugsFromSrc Should return an array of slugs extracted from data-src 1`] = `
Array [
  "image_11857",
  "image_11858",
  "image_11861",
  "image_11859",
  "image_11860",
]
`;

exports[`Helper getSlugsFromSrc Should return an array of slugs extracted from src 1`] = `
Array [
  "image_11857",
  "image_11858",
  "image_11861",
  "image_11859",
  "image_11860",
]
`;

exports[`Helper getSlugsFromSrc Should return an empty array if no slugs are extracted 1`] = `Array []`;

exports[`Helper getSlugsFromSrc Should return the right slugs if the filename has a size 1`] = `
Array [
  "image_11857",
  "image_11858x",
  "image_11861-",
  "image_11859_",
  "image_11860---",
]
`;
